### PR TITLE
add unit tests for url.c

### PR DESCRIPTION
**Pull request by xquery**
add some unit tests for url.c , remove stale code in url.h

### CODE REVIEW

File: lib/url.h [hunk 1]
```diff
@@ -34,7 +34,7 @@
 CURLcode Curl_init_do(struct Curl_easy *data, struct connectdata *conn);
 CURLcode Curl_open(struct Curl_easy **curl);
 CURLcode Curl_init_userdefined(struct Curl_easy *data);
-CURLcode Curl_dupset(struct Curl_easy * dst, struct Curl_easy * src);
```
Comment (danielgustafsson) [inline lib/url.h]: `Curl_dupset()` is referred to in a comment in `lib/urldata.h` as well which should be cleanup along with this removal.

File: tests/unit/unit1620.c [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+  rc = Curl_connect(empty, &empty->easy_conn, &async, &protocol_connect);
+  fail_unless(rc == CURLE_URL_MALFORMAT,
+              "Curl_connect() failed to return CURLE_URL_MALFORMAT");
+
+  fail_unless(empty->magic == CURLEASY_MAGIC_NUMBER,
+              "empty->magic should be equal to CURLEASY_MAGIC_NUMBER");
+
+  rc = Curl_connect(empty, &empty->easy_conn, &async, &protocol_connect);
+  fail_unless(rc == CURLE_URL_MALFORMAT,
+              "Curl_connect() failed to return CURLE_URL_MALFORMAT");
```
Comment (danielgustafsson) [inline tests/unit/unit1620.c]: Is the duplicated test serving a useful test purpose? If so, a comment would helpful to readers.

File: lib/url.h (hunk 1 above)
Comment (xquery) [reply lib/url.h]: nice catch, will remove

File: tests/unit/unit1620.c (hunk 2 above)
Comment (xquery) [reply tests/unit/unit1620.c]: yes it is intentional eg. doing a double invoke is an attempt to 'smoke out' any kind of assumptions (on state) that this func will only ever be invoked once (with clean up afterwards) ... thank you for highlighting, will add some comments.

**Comment by xquery** [comment]
going to close and re-open new PR ...

**Pull request closed without merging**